Iniciar laberinto

marivgil opened the pull request.

marivgil committed:
se agrega id de usuario y laberinto

marivgil committed:
servicio iniciar laberinto base

marivgil committed:
-se agregan ids necesarios a los sujetos del dominio
-se crea nueva excepcion
-se agrega volumen al repo
-se corrigen validaciones y cacheos

marivgil committed:
- se crean minificados para no enviar datos de mas
- se agrega una excepcion de usuario inexistente
- se adapta repo a minificados
- se crea paquete para controllers

agusgs commented on src/main/java/ar/edu/unq/ciu/controllers/IniciarLaberintoController.xtend:
```diff
@@ -0,0 +1,61 @@
+		try{
+			validarExisteUsuario(idUsuario)
+			validarExisteLaberintoParaUsuario(idUsuario, idLaberinto)
+			ok(RepoUsuarios.getInstance.buscarLaberinto(idUsuario,idLaberinto).toJson)
```
Las validaciones deberíamos meterlas directamente en el repo de objetos. En general la responsabilidad del controller deberia ser solo de formatear datos para enviar y recibir.

agusgs commented on src/main/java/ar/edu/unq/ciu/controllers/IniciarLaberintoController.xtend:
```diff
@@ -0,0 +1,61 @@
+		ok(usuarios.toJson)
+	}
+	*/
+	
```
y ese codigo comentado de que es?

marivgil committed:
Merge branch 'master' of https://github.com/agusgs/Gato-Encerrado-Dominio into iniciarLaberinto

marivgil committed:
- se unifican controllers
- se implementa minificador
- se adapta repo a la imple del minificador
- se elimina usuarioMini, ya no es necesario

agusgs commented on pom.xml:
```diff
@@ -74,5 +74,12 @@
+			<artifactId>xtrest</artifactId>
+			<version>0.1.4</version>
+		</dependency>
+		
```
Mariu fijate que quedo dos veces la dependencia de xtrest ;)

agusgs commented on src/main/java/ar/edu/unq/ciu/controllers/LaberintosController.xtend:
```diff
@@ -42,6 +46,39 @@ class LaberintosController {
+		try{
+			validarExisteUsuario(idUsuario)
+			validarExisteLaberintoParaUsuario(idUsuario, idLaberinto)
+			ok(iniciarLaberintoBusquedaLabYUser(idUsuario,idLaberinto).toJson)
```
estas validaciones iban al repo de objetos.

marivgil committed:
- correccion duplicidad pom (malditas decisiones de git)
- se nueven validaciones al repo
